text/0000-fusion-release-process.md

```diff
@@ -0,0 +1,71 @@
+* Start Date: 2018-02-06
+* RFC PR: (leave this empty)
+* React Issue: (leave this empty)
+
+# Summary
+
+A better process for release quality verification and publishing for the FusionJS core ecosystem.
+
+# Motivation
+
+Currently FusionJS packages live within multiple repositories and it's hard to tell the state of the FusionJS core ecosystem. We often release breaking changes in one package, but don't get around to upgrading dependent packages, or are aware of cross-package breakage that occurs. Our lack of cross-repo verification often leads to a fragmented and broken ecosystem, with no signal as to the overall current state.
+
+The goal of this RFC is to define a verification and release process, certifying FusionJS release quality, and allowing for faster and higher quality releases.
+
+# Detailed design
+
+### On-demand monorepo construction
+
+We will continue to develop individual FusionJS plugins and libraries in individual repositories. This gives us fast development cycles and avoids monorepo pain points like long boostrap time. A new CI job will exist which will be triggered for every commit going into master across all repositories. This job will construct a monorepo and run tests against the latest HEAD for all FusionJS repositories.
+
+We will have the best of a multirepo world, along with tooling that gives us the benefits of a monorepo.
+
+### Topologically sorted dependency graph
+
+We build a topologically sorted, batched graph of dependencies which will be used to build modules in parallel. FusionJS plugins will be transpiled, and copied into every dependent module. Non-fusion dependencies will continued to be installed through yarn. We choose to copy modules rather than symlink as that's more representative to an end-user using FusionJS, and also works better with static analysis.
```

Comment:
Suppose we need to create a patch fix to an older release of some package that has other FusionJS dependencies. In that case, how would these FusionJS dependencies be installed (since installing from HEAD would yield newer versions than that point in time). Tags or something?

Reply:
I think in this case you would branch off at the older version, and release from that branch. Each release should probably have a tag as well? Open to suggestions/ideas though.